### PR TITLE
ASU 432 | Refactor apartment application business logic

### PR DESCRIPTION
**Pull request by klemmari1**
- Add missing fields to factories and the serializers.
- Add `blank=True` for the `rejection_description` because it can be blank when an application is not rejected yet.
- Fix typos in tests
- Check the apartment uuids in tests.

Reformat code using [Django-Styleguide](https://github.com/HackSoftware/Django-Styleguide)
- Move business logic mostly to `models.py`, `selectors.py` and
`services.py`.
- Move validation to `clean()` and run `full_clean()` when saving.

### CODE REVIEW

**Review by ehaivala — CHANGES_REQUESTED** [review]
Sorry for the really nitpicky comments. 😬 

The comment about annotations is more of PSA instead of a thing that needs to be changed. It's just something to keep in mind if you weren't aware of it and want to use more detailed annotations at some point. 🙂

File: application_form/selectors.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+            apartment=apartment,
+        )
+        .order_by("haso_application__right_of_occupancy_id")
+        .values_list("haso_application__right_of_occupancy_id", flat=True)
+    )
+
+
+def get_haso_apartment_uuids(haso_application: HasoApplication) -> list:
+    """
+    From the given Haso applicaiton, get all apartment uuids
```
[the remark this answers is not in the document]
Comment (klemmari1) [reply application_form/selectors.py:37]: ah, it just keeps happening :D

File: application_form/services.py [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+        )
+
+        priorities_updated = False
+        for haso_application in haso_applications:
+            application_priority_queryset = (
+                get_winning_haso_apartment_priorities_for_application(
+                    haso_application, winning_apartment_priorities
+                )[
+                    1:
+                ]  # Slice off the lowest `priority_number`.
```
[the remark this answers is not in the document]
Comment (klemmari1) [reply application_form/services.py:90]: Ah, yep. This used to look different at some point but I guess black messed it up :D I'll change.

File: application_form/selectors.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+
+def list_haso_applications() -> QuerySet:
+    """
+    Get all HasoApplications
+    """
+    return HasoApplication.objects.all()
+
+
+def get_apartment_haso_application_id_queue(apartment: Apartment) -> list:
```
[the remark this answers is not in the document]
Comment (klemmari1) [reply application_form/selectors.py:20]: Good point! I'll change that.

**Review by ehaivala — APPROVED** [review]
Great stuff 👍

**Pull request merged**